common/src/Microsoft.Azure.IIoT.Core/src/Hub/Client/IoTHubServiceHttpClient.cs

```diff
@@ -69,15 +69,42 @@ public sealed class IoTHubServiceHttpClient : IoTHubHttpClientBase,
                 // First try create device
                 try {
                     var device = NewRequest($"/devices/{twin.Id}");
-                    _serializer.SerializeToRequest(device, new {
-                        deviceId = twin.Id,
-                        capabilities = twin.Capabilities
-                    });
+                    if (string.IsNullOrEmpty(twin.ModuleId) && !string.IsNullOrEmpty(twin.DeviceScope)) {
+                        _serializer.SerializeToRequest(device, new {
+                            deviceId = twin.Id,
+                            capabilities = twin.Capabilities,
+                            deviceScope = twin.DeviceScope
+                        });
+                    }
+                    else {
+                        _serializer.SerializeToRequest(device, new {
+                            deviceId = twin.Id,
+                            capabilities = twin.Capabilities
+                        });
+                    }
                     var response = await _httpClient.PutAsync(device, ct);
                     response.Validate();
                 }
-                catch (ConflictingResourceException)
-                    when (!string.IsNullOrEmpty(twin.ModuleId) || force) {
+                catch (ConflictingResourceException) {
+                    // try update the device scope if provided and not iotedge device and not a module
+                    if ((twin.Capabilities?.IotEdge).GetValueOrDefault(false) && 
```

Comment:
if twin.Capabilities?.IotEdge is null, this will generate an exception

Reply:
If twin itself is null, indeed it generates an exception, but this cannot happen since twin argument is validated in the beginning of the method.  GetValueOrDefault(false) will make sure the end result is false in case either Capabilities or IoTEdge are null.

Reply:
This changes current behavior for everything using patch.  My preference would be to leave the original code and specify "force = true".